tests/unit/tools/test_tools_utils.py

```diff
@@ -149,3 +149,240 @@ def test_authorlist_ignores_space_between_authors_and_affiliations():
+    )
+
+    expected = (
+        '100__ $$aAli, Y.X.$$vCERN $$vDESY\n'
+        '700__ $$aAndronov, E. I.$$vDESY'
+    )
+    result = authorlist(text)
+
+    assert expected == result.encode('utf-8')
+
+
+def test_authorlist_affiliation_with_numbers_and_letters():
+    text = (
+        'O. Buchmueller\n'
+        '1\n'
+        'K. Agashe\n'
+        '2\n'
+        '\n'
+        '1.\n'
+        'High Energy Physics Group, Blackett Laboratory, Imperial College, '
+        'Prince Consort Road, London SW7 2AZ, UK\n'
+        '2.\n'
+        'University of Maryland, Department of Physics, College Park, '
+        'MD 20742-4111, USA\n'
+    )
+
+    expected = (
+        '100__ $$aBuchmueller, O.$$vHigh Energy Physics Group, Blackett '
+        'Laboratory, Imperial College, Prince Consort Road, London SW7 2AZ, UK\n'
+        '700__ $$aAgashe, K.$$vUniversity of Maryland, Department of Physics, '
+        'College Park, MD 20742-4111, USA'
+    )
+    result = authorlist(text)
+
+    assert expected == result.encode('utf-8')
+
+def test_authorlist_note_symbols():
+    """Test authors which have some footnote symbols like † and ∗"""
+    text = (
+        'Y.X. Ali†1, 20, E I Andronov20∗\n'
+        '1 CERN\n'
+        '20 DESY'
+    )
+
+    expected = (
+        '100__ $$aAli, Y.X.$$vCERN $$vDESY\n'
+        '700__ $$aAndronov, E. I.$$vDESY'
+    )
+    result = authorlist(text)
+
+    assert expected == result.encode('utf-8')
+
+def test_authorlist_comma_wrong_position():
+    """Test case when there is comma before affiliation id."""
+    text = (
+        'Y. Bao,\n'
+        '1\n'
+        'and A. Lambrecht,\n'
+        '1,\n'
+        '2\n'
+        '1\n'
+        'Department of Physics, University of Florida, Gainesville, Florida 32611\n'
+        '2\n'
+        'Laboratoire Kastler–Brossel, CNRS, ENS, Universit ́e Pierre et Marie '
+        'Curie case 74, Campus Jussieu, F-75252 Paris Cedex 05, France\n'
+    )
+    expected = (
+        '100__ $$aBao, Y.$$vDepartment of Physics, University of Florida, '
+        'Gainesville, Florida 32611\n'
+        '700__ $$aLambrecht, A.$$vDepartment of Physics, University of '
+        'Florida, Gainesville, Florida 32611 $$vLaboratoire Kastler-Brossel, '
+        'CNRS, ENS, Universit ́e Pierre et Marie Curie case 74, Campus Jussieu, '
+        'F-75252 Paris Cedex 05, France'
+    )
+    result = authorlist(text)
+
+    assert expected == result.encode('utf-8')
+
+
+def test_authorlist_when_aff_line_ends_in_number():
+    text = (
+        'T.M. Liss\n'
+        '91\n'
+        'L. Littenberg\n'
+        '92\n'
+        '91.\n'
+        'Division of Science, City College of New York, 160 Convent Avenue, '
+        'New York, NY 10031\n'
+        '92.\n'
+        'Physics Department, Brookhaven National Laboratory, Upton, NY 11973, USA'
+    )
+
+    expected = (
+        '100__ $$aLiss, T.M.$$vDivision of Science, City College of New York, '
+        '160 Convent Avenue, New York, NY 10031\n'
```

Comment:
It's better to put these two lines together, otherwise this becomes a game of "spot the `\n`":
```
expected = (
    '100__ $$aLiss, T.M.$$vDivision of Science, City College of New York, 160 Convent Avenue, New York, NY 10031\n'
    '700__ $$aLittenberg, L.$$vPhysics Department, Brookhaven National Laboratory, Upton, NY 11973, USA'
)
```
(nobody cares about the 80 characters limit)